### PR TITLE
Refactor file generation process

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Refactors file generation process

**Special notes for your reviewer**:

_**WIP**_: I have some ideas about refactoring zip file generation too. Will take a look at that tomorrow.

### CODE REVIEW

**Comment by erbesharat** [comment]
Forgot to rewrite the tests, will fix it later.

**Comment by stp-ip** [comment]
cc: @ArmanMazdaee

**Comment by stp-ip** [comment]
Ready for review even with the failing tests?

File: generator/generator.go [hunk 1]
```diff
@@ -29,105 +30,112 @@ func createDir(storagePath string, filePath string) error {
 		if err != nil {
 			return fmt.Errorf("unable to create path: %s", storagePath+path)
 		}
+		return nil
```
Comment (ArmanMazdaee) [inline generator/generator.go:33]: I believe this line is unnecessary.

File: generator/generator.go [hunk 2]
```diff
@@ -10,14 +10,15 @@ import (
...
 	"github.com/okkur/reposeed/cmd/reposeed/config"
 	templates "github.com/okkur/reposeed/cmd/reposeed/templates"
 	"github.com/rs/xid"
 )
 
 type JSONerror struct {
-	Code    int    `json:"code"`
-	Message string `json:"message"`
+	Code    int
+	Message string
```
Comment (ArmanMazdaee) [inline generator/generator.go]: I think we need the struct tags.

**Comment by erbesharat** [comment]
It's failing because of outdated dependencies. I don't know why but we have reposeed-server in our vendor too and it's ``okkur/reposeed-server`` which is the outdated version.

**Review by ArmanMazdaee — CHANGES_REQUESTED** [review]
I think everything is fine beside the mentioned issues.

File: generator/generator.go (hunk 1 above)
Comment (erbesharat) [reply generator/generator.go:33]: If we remove this line we get ``missing return at end of function`` error.

**Comment by erbesharat** [comment]
@stp-ip I think it's ready to get merged.

**Pull request merged**